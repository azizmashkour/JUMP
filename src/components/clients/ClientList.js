import React, { useState } from 'react'
import PropTypes from 'prop-types';
import { withTranslation } from 'react-i18next';
import FakeClients from '../../data/FakeClients';
import ClientDetails from './ClientDetails';
import './ClientStyles.css';

const ClientList = (props) => {
  const [state, setState] = useState({ address: null });

  const handleCheckaddress =(e)=> {
      setState({ address: e.target.value });
  }

  return (
    <div className="table-container">
      <>
        <h1 className="text-center">{props.t('client_list.clients_list')}</h1>
        <div className="table-responsive">
          <table className="table table-striped table-bordered table-light">
            <thead>
              <tr>
                <th></th>
                <th>{props.t('client_list.last_name')}</th>
                <th>{props.t('client_list.first_name')}</th>
              </tr>
            </thead>
            <tbody>
              {
                FakeClients.map((client, index) => {
                  return (
                    <tr key={index}>
                      <td className="text-center">
                        <input className="client-check" id={client.id} type="radio" name="id" value={client.address} onClick={(e) =>handleCheckaddress(e)}/>
                      </td>
                      <td>{client.nom}</td>
                      <td>{client.prenom}</td>
                    </tr>
                  )
                })
              }
            </tbody>
          </table>
        </div>
      </>
      {state.address != null && <ClientDetails address={state.address}/>}
    </div>
  );
};

ClientList.propTypes = {
  t: PropTypes.func.isRequired,
};

export default withTranslation()(ClientList);
