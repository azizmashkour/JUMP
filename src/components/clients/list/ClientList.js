import React, { useRef, useState, useEffect } from 'react'
import PropTypes from 'prop-types';
import { withTranslation } from 'react-i18next';
import FakeClients from '../../../data/FakeClients';
import ClientDetails from '../details/ClientDetails';
import '../ClientStyles.css';

const ClientList = (props) => {
  const [state, setState] = useState({
    address: null,
    activeClientId: null,
    newData: [],
    noResult: false,
  });

  const hasMount = useRef(false);

  useEffect(() => {
    if (!hasMount.current) {
      // componentDidMount
      setState({newData: FakeClients})
    } else {
      // componentDidUpdate
    }
    return () => {
      // componentWillUnmount
    };
  }, []);

  const handleCheckaddress =(address, clientId)=> {
    setState({
      newData: FakeClients,
      address: address,
      activeClientId: clientId,
    });
  }

  // Search typed keywords from user first and last name,
  // then update the initial clients list state and,
  // display error message when there is no client found with the search query
  const handleSearch =(e)=> {
    const typedWords = e.target.value;
    if (typedWords.trim() !== "") {
      const stateRes =  FakeClients
                          .filter((option) =>
                            option.prenom.toLowerCase().includes((typedWords).toLowerCase()) ||
                            option.nom.toLowerCase().includes((typedWords).toLowerCase()))
                              .map(filteredPerson => (filteredPerson));

      stateRes.length > 0 ? setState({newData: stateRes, noResult: false}) : setState({newData: [], noResult: true});
    } else {
      setState({newData: FakeClients});
    }
  }

  return (
    <div className="table-container">
      <>
        <h1 className="text-center">{props.t('client_list.clients_list')}</h1>
        <form>
          <div className="form-control mb-0">
            <input
              autoComplete="off"
              type="text"
              id="searchClient"
              placeholder={props.t('client_list.search_client')}
              onChange={(e)=>handleSearch(e)}
            />
          </div>
        </form>

        {state.noResult && <div className="text-danger">{props.t('client_list.no_client_found')}</div>}

        <div className="table-responsive">
          <table className="table table-striped table-bordered table-light">
            <thead>
              <tr>
                <th></th>
                <th>{props.t('client_list.last_name')}</th>
                <th>{props.t('client_list.first_name')}</th>
              </tr>
            </thead>
            <tbody>
              {
                state.newData && state.newData.map((client, index) => {
                  return (
                    <tr key={index} className={client.id === state.activeClientId ? "active_client" : ""}
                      onClick={() => handleCheckaddress(client.address, client.id)}>
                      <td className="text-center">
                        <label htmlFor={client.id}>
                          <input className="client-check" id={client.id} type="radio" name="id"
                            checked={client.id === state.activeClientId} readOnly/>
                        </label>
                      </td>
                      <td>{client.prenom}</td>
                      <td>{client.nom}</td>
                    </tr>
                  )
                })
              }
            </tbody>
          </table>
        </div>
      </>
      {state.address != null && <ClientDetails address={state.address}/>}
    </div>
  );
};

ClientList.propTypes = {
  t: PropTypes.func.isRequired,
};

export default withTranslation()(ClientList);
