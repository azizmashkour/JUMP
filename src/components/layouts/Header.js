import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { withTranslation } from 'react-i18next';
import BrowserLanguage from '../../utils/BrowserLanguage';
import Select from 'react-select';
import { options, langs } from '../../configs/options';

const Header = ({lang, i18n, t}) => {
  const [state, setState] = useState({
    defaultLang: {
      label: i18n.language === 'en' ?  langs.en: langs.fr,
      value: i18n.language === 'en' ? langs.en : langs.fr,
    }
  });

  const changeLang = (lang) => {
    if (!lang) {
      const tempLang = BrowserLanguage.getDefaultLanguage();
      lang = tempLang === 'en' ? 'fr' : 'en';
    }
    setState({ defaultLang: lang });

    if (BrowserLanguage.setLanguage(lang.value)) {
      // Reload page if browser support localStorage
      window.location.reload(); // Relaod app after langue change
    } else {
      // Change language at runtime if localStorage not found
      i18n.changeLanguage(lang); // Need for change language at runtime
    }
  };

  return (
    <div className="header-container">
      <Select
        defaultValue={state.defaultLang}
        options={options}
        value={state.defaultLang}
        onChange={changeLang}
        className="App-Select"
        isSearchable={false}
      />
    </div>
  );
};

Header.propTypes = {
  t: PropTypes.func.isRequired,
  i18n: PropTypes.shape({
    defaultNS: PropTypes.string,
    changeLanguage: PropTypes.func,
  }).isRequired,
};

export default withTranslation()(Header);
