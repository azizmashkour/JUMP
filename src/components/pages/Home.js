import React, { useState } from 'react';
import { withTranslation } from 'react-i18next';
const Home = ({lang, i18n, t}) => {

  return (
    <div className="container">
    </div>
  );
};

export default withTranslation()(Home);
