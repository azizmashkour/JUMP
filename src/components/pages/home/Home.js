import React from 'react';
import Header from '../../layouts/Header';
import ClientList from '../../clients/list/ClientList';

const Home = () => {

  return (
    <>
      <Header />
      <ClientList/>
    </>
  );
};

export default Home;
