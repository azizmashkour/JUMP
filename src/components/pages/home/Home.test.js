import React from "react";
import { shallow } from "enzyme";
import Home from "./Home";
import Header from '../../layouts/Header';
import ClientList from '../../clients/list/ClientList';

describe('<Home />', () => {
  it("should display home component", () => {
    const wrapper = shallow(<Home />);
    expect(wrapper.find(Header)).toBeTruthy();
    expect(wrapper.find(ClientList)).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
