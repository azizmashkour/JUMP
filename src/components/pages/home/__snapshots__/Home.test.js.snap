// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should display home component 1`] = `ShallowWrapper {}`;
